public/app/features/alerting/unified/components/receivers/form/fields/TemplateContentAndPreview.tsx

```diff
@@ -0,0 +1,100 @@
+import { css, cx } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Box, useStyles2 } from '@grafana/ui';
+import { useAlertmanager } from 'app/features/alerting/unified/state/AlertmanagerContext';
+import { GRAFANA_RULES_SOURCE_NAME } from 'app/features/alerting/unified/utils/datasource';
+
+import { EditorColumnHeader } from '../../../contact-points/templates/EditorColumnHeader';
+import { getPreviewResults } from '../../TemplatePreview';
+import { usePreviewTemplate } from '../../usePreviewTemplate';
+
+export function TemplateContentAndPreview({
+  payload,
+  templateContent,
+  templateName,
+  payloadFormatError,
+  setPayloadFormatError,
+  className,
+}: {
+  payload: string;
+  templateName: string;
+  payloadFormatError: string | null;
+  setPayloadFormatError: (value: React.SetStateAction<string | null>) => void;
+  className?: string;
+  templateContent: string;
+}) {
+  const styles = useStyles2(getStyles);
+
+  const { selectedAlertmanager } = useAlertmanager();
+  const isGrafanaAlertManager = selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+
+  const { data } = usePreviewTemplate(templateContent, templateName, payload, setPayloadFormatError);
+  const previewToRender = getPreviewResults(undefined, payloadFormatError, data);
+
+  return (
+    <div className={cx(styles.container, className)}>
+      <EditorColumnHeader label="Template content" />
+      <Box flex={1}>
+        <div className={styles.viewerContainer({ height: 400 })}>
+          <div className={styles.templateContent}>{templateContent}</div>
+        </div>
+      </Box>
+
+      {isGrafanaAlertManager && (
```

Comment:
for non-Grafana Alertmanagers we only let users selecting the template and see its content but not previewing